Azure-Samples/chat-with-your-data-solution-accelerator
# Add multi-language support for speech input on web.

## Purpose
Required by #549. Without these changes, the chat frontend only recognizes english.  This PR adds support for multi-lingual speech recognition from the Chat frontend. Currently the additional languages are hardcoded into the frontend code, however, the plan under #317 is to move this list of languages to the deployment configuration in the near future. 

This PR also introduces the first frontend unit test with the required scaffolding and support to extend this test suite. Hereafter the expectation for frontend changes will be that they are accompanied by one or more unit tests. 


## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Deploy CWYD, and speak through the microphoine input in one of the following languages: French, German, Italian. Your words should be converted correctly to text. 

To run the frontend unit tests, checkout the code and run 
```
make unittest-frontend
```
